firestore-counter/clients/dart/distributed_counter.dart

```diff
@@ -0,0 +1,95 @@
+import 'dart:async';
+
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+const SHARD_COLLECTION_ID = '_counter_shards_';
+
+class DistributedCounter {
+  static String shardId;
```

Comment:
Why the static state? The TS example doesn't have such `init` function either

Reply:
The shardId is effectively static since in the TS version it's derived from a cookie.